api/src/main/java/com/redhat/insights/tls/PEMSupport.java

```diff
@@ -96,14 +96,19 @@ SSLContext createTLSContext(byte[] certPemData, byte[] keyPemData) {
     char[] keystorePassword = new char[0];
     try {
 
-      Certificate[] certificates =
-          parsePemData(Certificate.class, certPemData).toArray(new Certificate[0]);
+      X509Certificate[] certificates =
+          parsePemData(X509Certificate.class, certPemData).toArray(new X509Certificate[0]);
       List<PrivateKey> keys = parsePemData(PrivateKey.class, keyPemData);
 
       KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
       keyStore.load(null);
 
       for (int i = 0; i < certificates.length; i++) {
+        try {
+          certificates[i].checkValidity();
+        } catch (CertificateException e) {
```

Comment:
I don't think you have to do a `catch` here since there's one for a `CertificateException` below.

Reply:
Also there's a test failure in CI (time not matching)

Reply:
timezones :)